spark/src/main/scala/geotrellis/spark/io/cassandra/CassandraDriver.scala

```diff
@@ -0,0 +1,54 @@
+package geotrellis.spark.io.cassandra
+
+import java.nio.ByteBuffer;
+
+import geotrellis.spark._
+import geotrellis.raster._
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+import com.datastax.spark.connector.rdd.CassandraRDD
+import com.datastax.spark.connector._
+
+import com.datastax.driver.core.DataType.{text, blob}
+import com.datastax.driver.core.schemabuilder.SchemaBuilder
+import com.datastax.driver.core.Session
+
+class TableNotFoundError(table: String) extends Exception(s"Target Cassandra table '$table' does not exist.")
+
+trait CassandraDriver[K] extends Serializable {
+  def applyFilter(rdd: CassandraRDD[(String, ByteBuffer)], layerId: LayerId, filterSet: FilterSet[K]): RDD[(String, ByteBuffer)]
+  def decode(rdd: RDD[(String, ByteBuffer)], metaData: RasterMetaData): RasterRDD[K]
+  def rowId(id: LayerId, key: K): String
+
+  def load(sc: SparkContext, keyspace: String)(
+    id: LayerId, metaData: RasterMetaData, table: String, filters: FilterSet[K]): RasterRDD[K] = {
+
+    val rdd: CassandraRDD[(String, ByteBuffer)] = 
+      sc.cassandraTable[(String, ByteBuffer)](keyspace, table).select("id", "tile")
+
+    val filteredRDD = applyFilter(rdd, id, filters)
```

Comment:
I may have to eat my words on this one. Looking closer at it `CassandraRDD.where` just appends the clause and `UnionRDD` does not appear to cause a collect. This may actually be doing what we want. It would be great to ingest a large layer and see the difference in the fetch of a small query extent vs a large one. Also it would be interesting to see how it behaves when there are thousands of where clauses with a large enough query extent.